src/core/symbology/qgsmarkersymbollayer.cpp

```diff
@@ -2984,6 +2985,8 @@ QgsSymbolLayer *QgsFontMarkerSymbolLayer::create( const QgsStringMap &props )
 
   QgsFontMarkerSymbolLayer *m = new QgsFontMarkerSymbolLayer( fontFamily, string, pointSize, color, angle );
 
+  if ( props.contains( QStringLiteral( "font_style" ) ) )
+    m->setFontStyle( props[QStringLiteral( "font_style" )] );
```

Comment:
Can you checkout how the text renderer stores this? From memory there's some trickiness involving translation.